src/internal/github/assets.go

```diff
@@ -1,14 +1,51 @@
 package github
 
 import (
+	"context"
+	"fmt"
+	"io"
+	"log"
+	"net/http"
 	"regexp"
+	httpInternal "registry-stable/internal/http"
 )
 
 type Platform struct {
 	OS   string
 	Arch string
 }
 
+func DownloadAssetContents(ctx context.Context, downloadURL string) (body io.ReadCloser, err error) {
+	httpClient := httpInternal.GetHttpRetryClient()
+
+	log.Printf("Downloading asset, url: %s", downloadURL)
+	req, reqErr := http.NewRequestWithContext(ctx, http.MethodGet, downloadURL, nil)
+	if reqErr != nil {
+		log.Printf("Failed to create request %s", reqErr)
+		err = fmt.Errorf("failed to create request: %w", reqErr)
+		return
+	}
+
+	resp, respErr := httpClient.Do(req)
+	if respErr != nil {
+		log.Printf("Error downloading asset %s", respErr)
+		err = fmt.Errorf("error downloading asset: %w", respErr)
+		return
+	}
+
+	if resp.StatusCode != http.StatusOK {
+		resp.Body.Close()
+		log.Printf("Unexpected status code when downloading asset: %d", resp.StatusCode)
+		err = fmt.Errorf("unexpected status code when downloading asset: %d", resp.StatusCode)
+		return
+	}
+
+	body = resp.Body
```

Comment:
I feel like this code has come across from the old registry,and I was not a fan of it back then because the caller has to be aware that they need to be closing the body. 

Maybe we could now have this return a string/[]byte instead?

Reply:
Done in 8abba6412fe3fb6d26eee823a5d2e25f3b55a626